src/pip/_internal/pyproject.py

```diff
@@ -89,9 +90,15 @@ def load_pyproject_toml(
 
     # If we haven't worked out whether to use PEP 517 yet,
     # and the user hasn't explicitly stated a preference,
-    # we do so if the project has a pyproject.toml file.
+    # we do so if the project has a pyproject.toml file
+    # or if we cannot import setuptools.
+
+    # We fallback to PEP 517 when without setuptools,
+    # so setuptools can be installed as a default build backend.
+    # For more info see:
+    # https://discuss.python.org/t/pip-without-setuptools-could-the-experience-be-improved/11810/9
     elif use_pep517 is None:
-        use_pep517 = has_pyproject
+        use_pep517 = has_pyproject or not importlib.util.find_spec("setuptools")
```

Comment:
Note that the setuptools check can be technically made much sooner in this function, but this way, it is avoided when not necessary. Not sure which approach is better.